packages/synthetix-main/contracts/mixins/MarketManagerMixin.sol

```diff
@@ -139,7 +139,24 @@ contract MarketManagerMixin is MarketManagerStorage, PoolModuleStorage {
         return IMarket(marketData.marketAddress).reportedDebt();
     }
 
+    function _depositedCollateralValue(MarketData storage marketData) internal view returns (uint) {
+        uint totalDepositedCollateralValue = 0;
+        address[] memory collateralTypes = _collateralStore().collateralTypes.values();
```

Comment:
rather than iterating all the collateral types, it would be better to iterate over a short list of *collateral entries* (aka, collateral address/amount pair in a struct). Since most markets will only work with one/a small subset of collatreal type in our stated use case